ControlFlowAnalyser: Also consider called functions in a flow (#10143)
ethereum/solidity: Changelog.md

Bugfixes:
 * SMTChecker: Fix lack of reporting potential violations when using only the CHC engine.
 * Code generator: Fix missing creation dependency tracking for abstract contracts.

>  * Control Flow Graph: Remove wrong warning when functions that always ``revert()`` are called

Reviewer: ```suggestion
 * Control Flow Graph: Remove wrong warning when a function has branches that always revert.
```

Bugfixes:

>  * Control Flow Graph: Take internal calls to functions that always revert into account.

Reviewer 2: ... for reporting unused or unassigned variables.

 * SMTChecker: Fix internal error in the BMC engine when inherited contract from a different source unit has private state variables.
 * SMTChecker: Fix internal error when ``array.push()`` is used as the LHS of an assignment.
 * Code generator: Fix missing creation dependency tracking for abstract contracts.

>  * Control Flow Graph: Remove wrong warning when a function has branches that always revert.

Reviewer 2: "Take internal calls to functions that always revert into account"?

 * SMTChecker: Fix internal error on try-catch with function call in catch block.
 * Type Checker: Fix missing error when events are used without an emit statement.
 * Commandline interface: Fix internal error when printing AST and using ``--base-path`` or ``file://`` prefix in imports.

>  * Control Flow Graph: Take internal calls to functions that always revert into account.

Reviewer 2: Should be further up (changelog is sorted alphabetically)

 * Yul Optimizer: Fix a bug in NameSimplifier where a new name created by NameSimplifier could also be created by NameDispenser.
 * Yul Optimizer: Removed NameSimplifier from optimization steps available to users.

> Bug Fixes:

Reviewer 2: Totally wrong place :)
